File: backend/app/services/post_code_service.py
import requests
import time
from urllib.parse import quote

def get_postcode(city, street="", number=""):
    try:
        url = "https://nominatim.openstreetmap.org/search"
        params = {
            "street": f"{number} {street}",
            "city": city,
            "country": "Poland",
            "format": "json",
            "addressdetails": 1,
            "limit": 1
        }
        headers = {"User-Agent": "my-teryt-app/1.0"}
        

        r = requests.get(url, params=params, headers=headers, timeout=10)
        data = r.json()
        if data:
            postcode = data[0].get("address", {}).get("postcode")
            if postcode:
                return postcode
    except Exception as e:
        print("Error Nominatim:", e)
    return None

File: frontend/src/services/postcodeService.ts
import axiosInstance from "../interceptor/interceptor";

export async function findPostcode(city: string, street: string, number: string): Promise<{ type: string, postcode?: string, postcodes?: string[] }> {
  const response = await axiosInstance.get("/api/postcode/search", {
    params: { city, street, number }
  });
  return response.data;
}

export async function findCityByPostcode(postcode: string): Promise<{ postcode: string, cities: string[] }> {
  const response = await axiosInstance.get(`/api/postcode/city/${postcode}`);
  return response.data;
}
